recipes/pango/all/conanfile.py

```diff
@@ -0,0 +1,140 @@
+    homepage = "https://www.pango.org/"
+    topics = ("conan", "fontconfig", "fonts", "freedesktop")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    generators = "pkg_config"
+    _source_subfolder = "source_subfolder"
+    _autotools = None
+
+    @property
+    def _is_msvc(self):
+        return self.settings.compiler == "Visual Studio"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def build_requirements(self):
+        if not tools.which("pkg-config"):
+            self.build_requires("pkgconf/1.7.3")
+        self.build_requires("meson/0.54.2")
+
+    def requirements(self):
+        if self.settings.os != "Windows":
+            self.requires("fontconfig/2.13.92")
+        if self.settings.os == "Linux":
+            self.requires("xorg/system")
+        self.requires("cairo/1.17.2")
+        self.requires("harfbuzz/2.7.2")
+        self.requires("glib/2.67.0")
+        self.requires("fribidi/1.0.9")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extrated_dir = self.name + "-" + self.version
+        os.rename(extrated_dir, self._source_subfolder)
+
+    def _configure_meson(self):
+        defs = dict()
+        defs["introspection"] = "disabled"
+        meson = Meson(self)
+        meson.configure(build_folder="build", source_folder=self._source_subfolder, defs=defs, args=['--wrap-mode=nofallback'])
+        return meson
+
+    def build(self):
+        meson_build = os.path.join(self._source_subfolder, "meson.build")
+        tools.replace_in_file(meson_build, "subdir('tests')", "")
+        tools.replace_in_file(meson_build, "subdir('tools')", "")
+        tools.replace_in_file(meson_build, "subdir('utils')", "")
+        tools.replace_in_file(meson_build, "subdir('examples')", "")
+        with tools.environment_append(VisualStudioBuildEnvironment(self).vars) if self._is_msvc else tools.no_op():
+            meson = self._configure_meson()
+            meson.build()
+
+    def package(self):
+        self.copy(pattern="COPYING", dst="licenses", src=self._source_subfolder)
+        with tools.environment_append(VisualStudioBuildEnvironment(self).vars) if self._is_msvc else tools.no_op():
+            meson = self._configure_meson()
+            meson.install()
+        if self.settings.compiler == "Visual Studio":
+            with tools.chdir(os.path.join(self.package_folder, "lib")):
+                for filename_old in glob.glob("*.a"):
+                    filename_new = filename_old[3:-2] + ".lib"
+                    self.output.info("rename %s into %s" % (filename_old, filename_new))
+                    shutil.move(filename_old, filename_new)
```

Comment:
any reason for not using os.rename? permission maybe? I'm curious because we don't have a consensus about it, but it would be great one

Reply:
Idk. We'd have to git blame the bincrafters recipe to know who wrote this!